src/main/java/com/example/solution/TasksService.java

```diff
@@ -0,0 +1,62 @@
+package com.example.solution;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class TasksService {
+    private final TasksRepository tasksRepository;
+
+    public Task create(Task tasks) {
+        return tasksRepository.save(tasks);
+    }
+
+    public List<Task> getAllDone() {
+        return new ArrayList<>(tasksRepository.findByDoneTrue());
+
+    }
+    public boolean getById(Long id) {
+        return tasksRepository.findById(id).isPresent();
+
+    }
+    public List<Task> getCategory(String category) {
+        return new ArrayList<>(tasksRepository.findByCategory(category));
+
+    }
+
+    public List<Task> getAllNotDone() {
+        return new ArrayList<>(tasksRepository.findByDoneFalse());
+
+    }
+
+    public void update(Long id, Task tasks) {
+        Optional<Task> optionalExistingTask = tasksRepository.findById(id);
+
+        if (optionalExistingTask.isPresent()) {
+            Task existingTask = optionalExistingTask.get();
+            if (!tasks.getName().isBlank()) {
+                existingTask.setName(tasks.getName());
+            }
+            if (!tasks.getCategory().isBlank()) {
+                existingTask.setCategory(tasks.getCategory());
+            }
+            if (tasks.getDueDate() != null) {
+                existingTask.setDueDate(tasks.getDueDate());
+            }
+            tasksRepository.save(existingTask);
```

Comment:
Wystarczy wywołać metodę save repozytorium. Jeżeli w obiekcie ustawione jest id, to spring data (hibernate pod spodem) wykona aktualizację